resilience4j-ratelimiter/src/main/java/io/github/resilience4j/ratelimiter/RateLimiter.java

```diff
@@ -185,7 +191,14 @@ static RateLimiter ofDefaults(String name) {
     ) {
         return () -> {
             waitForPermission(rateLimiter, permits);
-            return supplier.get();
+            try {
+                F result = supplier.get();
+                drainIfNeeded(rateLimiter, CallsResult.success(result));
```

Comment:
Please introduce and use an `onResult` hook

Reply:
where should I introduce it? in `RateLimiterEventProcessor` ?

if so then how should I pass the event to it from a static method? event processors are privet fields in rate limiters implementations

Reply:
See https://github.com/resilience4j/resilience4j/pull/1264#issuecomment-739824598

I'm talking about new public methods in the `RateLimiter` interface analog to the design of the CircuitBreaker, Bulkhead and Retry.

Reply:
done